From 2509e6c5c5d7e87353e981daec3a0a0d98eca415 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 23 Jun 2021 17:00:45 +0300
Subject: [PATCH] [SW-45099] common: use device id without vendor id

preboot_device_id_get() need to return the device id without the vendor
id. Failing to do so resulted with Gaudi-M initaliztion failure as it's
init proceute is different and detected by the device id.

Change-Id: I23f5fad1bc6388bea5237df3fe4ae644dd067f99
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/arch/gaudi/common.c b/arch/gaudi/common.c
index ebfe481..d371ac2 100644
--- a/arch/gaudi/common.c
+++ b/arch/gaudi/common.c
@@ -159,7 +159,7 @@
 	if ((pci_dev_id >> 16) != PCI_VENDOR_ID_HABANALABS)
 		return 0;
 
-	return pci_dev_id;
+	return pci_dev_id & 0xffff;
 }
 
 static void preboot_disable_pcie_ltr(void)
